### ISSUE TITLE
Creation of self-signed crt and combined leads to nginx errors

### ISSUE THREAD

**Issue by steve-gregory** [iPlantCollaborativeOpenSource/clank#79]
The error being reported:
`6571#6571: SSL_CTX_use_PrivateKey_file("/etc/ssl/private//self-signed.key") failed (SSL: error:0B080074:x509 certificate routines:X509_check_private_key:key values mismatch)`

The state of the file system:
```
root:~# ll /etc/ssl/private/se*
-rw-r--r-- 1 root root 1704 Jun 28 20:36 /etc/ssl/private/self-signed.key
root:~# ll /etc/ssl/certs/se*
-rw-r--r-- 1 root root 1350 Jun 28 20:35 /etc/ssl/certs/self_signed_combined.crt
-rw-r--r-- 1 root root 1350 Jun 28 20:36 /etc/ssl/certs/self-signed.crt
```
What was required to fix it:
```
cp /etc/ssl/certs/self-signed.crt /etc/ssl/certs/self_signed_combined.crt
```

**Comment by lenards** [comment]
Confirmed in vagrant environment.

**Comment by lenards** [comment]
You can also do:
```bash
rm /etc/ssl/certs/self_signed_combined.crt
cd /opt/dev/troposphere/extras/nginx
make /etc/ssl/certs/self_signed_combined.crt
```

**Comment by lenards** [comment]
This also affects celery & flower ..

**Comment by lenards** [comment]
I mentioned #73 because when you leave the following blank: 

```yaml
SSL_CERTIFICATE:    #Absolute Path Recommended
BUNDLE_CERT:        #Absolute Path Recommended
SSL_KEY:            #Absolute Path Recommended
```

These blank values will be used in `ATMO['celeryd.default']`. This seems to be one likely cause of the above message.

**Comment by lenards** [comment]
You can *see* this 👓 when looking at the output from `ps aux`:

```
# ps aux | grep flower
root      9590  0.0  2.1 621136 88164 pts/0    Sl   Jul01   6:12 python /opt/dev/atmosphere/manage.py celery flower --certfile=/etc/ssl/certs/ --keyfile=/etc/ssl/private/ --port=8443 
```

**Comment by lenards** [comment]
Remember - you need a `service atmosphere full-stop` to get flower to get these values. My experience within vagrant would seem to indicate that `service celeryd restart` is not going to restart _flower_ 🌸

**Comment by lenards** [comment]
The duplication "tasks" in #88 appear to cause the generated self_signed_combined.crt to represent the self-signed.crt & self-signed.key from calling `playbooks/deploy_atmosphere.yml` but the self_signed_combined.crt is **not** regenerated when `playbooks/deploy_troposphere.yml` runs.

**Comment by lenards** [comment]
Fixed by #102
